pinot-spi/src/main/java/org/apache/pinot/spi/trace/InvocationRecording.java

```diff
@@ -0,0 +1,113 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.trace;
+
+import org.apache.pinot.spi.data.FieldSpec;
+
+
+public interface InvocationRecording {
+
+  /**
+   * Sets the number of docs scanned by the operator.
+   * @param docsScanned how many docs were scanned.
+   */
+  default void setDocsScanned(int docsScanned) {
+  }
+
+  /**
+   * Sets the number of documents matching after a filter has been applied.
+   * Indicates whether the index was selective or not.
+   * @param numDocsMatchingAfterFilter how many docs still match after applying the filter
+   */
+  default void setNumDocsMatchingAfterFilter(int numDocsMatchingAfterFilter) {
```

Comment:
Bit confused between this and previous function. The current `numDocsScanned` metric that we measure and return actually represents the selectivity of the query. 

If `numDocsMatchingAfterFilter` (which I feel is a very intuitive name) represents selectivity, then what do the `docsScanned` in the previous function represent ? I guess totalDocs ?

Fair to say that `docsScanned` and `numDocsMatchingAfterFilter` will be same if no filter ?

Reply:
I'm trying to differentiate between usage of indexes and forward indexes, because they make different resource demands, which becomes increasingly important with tiered storage. If a filter uses an index, `numDocsScanned` should be zero for that operator, but `numDocsMatchingAfterFilter` will be positive. If you then project over that filter and hit the query limit < `numDocsMatchingAfterFilter`, you will have `numDocsScanned` = limit. If you do an aggregation over the filter, `numDocsScanned` = `numDocsMatchingAfterFilter`. If you have a `ScanBasedFilterOperator`, `numDocsScanned` = `numDocsMatchingAfterFilter` because it performed the filter by scanning the data.

I'm open to changing names to avoid clashing with existing concepts, or bringing the names in line with existing concepts if it makes this more useful for other SPI users.